pennylane/pauli/grouping/group_observables.py

```diff
@@ -177,6 +177,37 @@ def colour_pauli_graph(self):
         return self.grouped_paulis
 
 
+def _calculate_partitioned_coefficients(observables, partitioned_paulis, coefficients):
```

Comment:
Just extracting some code to get pylint to stop complaining.

Reply:
I did the same thing in https://github.com/PennyLaneAI/pennylane/pull/5559, you might need to merge from master and resolve some merge conflicts.